scoper.inc.php

```diff
@@ -173,99 +177,34 @@
-			$contents = str_replace( "\\$prefix\\wp_cache_get", '\\wp_cache_get', $contents );
-			$contents = str_replace( "\\$prefix\\set_transient", '\\set_transient', $contents );
-			$contents = str_replace( "\\$prefix\\wp_cache_set", '\\wp_cache_set', $contents );
-			$contents = str_replace( "\\$prefix\\wp_using_ext_object_cache", '\\wp_using_ext_object_cache', $contents );
-			$contents = str_replace( "\\$prefix\\_doing_it_wrong", '\\_doing_it_wrong', $contents );
-			$contents = str_replace( "\\$prefix\\plugin_dir_url", '\\plugin_dir_url', $contents );
-			$contents = str_replace( "\\$prefix\\is_admin_bar_showing", '\\is_admin_bar_showing', $contents );
-			$contents = str_replace( "\\$prefix\\get_bloginfo", '\\get_bloginfo', $contents );
-			$contents = str_replace( "\\$prefix\\add_filter", '\\add_filter', $contents );
-			$contents = str_replace( "\\$prefix\\remove_filter", '\\remove_filter', $contents );
-			$contents = str_replace( "\\$prefix\\apply_filters", '\\apply_filters', $contents );
-			$contents = str_replace( "\\$prefix\\add_query_arg", '\\add_query_arg', $contents );
-			$contents = str_replace( "\\$prefix\\remove_query_arg", '\\remove_query_arg', $contents );
-			$contents = str_replace( "\\$prefix\\get_post", '\\get_post', $contents );
-			$contents = str_replace( "\\$prefix\\wp_scripts", '\\wp_scripts', $contents );
-			$contents = str_replace( "\\$prefix\\wp_style_is", '\\wp_style_is', $contents );
-			$contents = str_replace( "\\$prefix\\WPMU_PLUGIN_DIR", '\\WPMU_PLUGIN_DIR', $contents );
-			$contents = str_replace( "\\$prefix\\WP_PLUGIN_DIR", '\\WP_PLUGIN_DIR', $contents );
-			$contents = str_replace( "\\$prefix\\WP_PLUGIN_URL", '\\WP_PLUGIN_URL', $contents );
-			$contents = str_replace( "\\$prefix\\WPMU_PLUGIN_URL", '\\WPMU_PLUGIN_URL', $contents );
-			$contents = str_replace( "\\$prefix\\wp_array_slice_assoc", '\\wp_array_slice_assoc', $contents );
-			$contents = str_replace( "\\$prefix\\wp_json_encode", '\\wp_json_encode', $contents );
-			$contents = str_replace( "$prefix\\WP_Http", 'WP_Http', $contents );
-			$contents = str_replace( "$prefix\\WP_Error", 'WP_Error', $contents );
-
-			return $contents;
-		},
-	],
+	'patchers'          => [],
 
 	// See https://github.com/humbug/php-scoper#whitelist.
-	'whitelist'                  => [],
+	'whitelist'         => [],
 
-	// See https://github.com/humbug/php-scoper#constants--classes--functions-from-the-global-namespace.
-	'whitelist-global-constants' => false,
+	'exclude-classes'   => $wp_classes,
 
-	// See https://github.com/humbug/php-scoper#constants--classes--functions-from-the-global-namespace.
-	'whitelist-global-classes'   => false,
+	'exclude-functions' => $wp_functions,
 
-	// See https://github.com/humbug/php-scoper#constants--classes--functions-from-the-global-namespace.
-	'whitelist-global-functions' => false,
+	'exclude-constants' => array_merge(
+		$wp_constants,
+		[
```

Comment:
Why are these needed? Don't they come from `$wp_constants`. If they are a temporary, what is the plan to get rid of them? If temporary, a comment / todo to fix these hardcoded values at a later date.

Reply:
Please see discussion at https://github.com/GoogleForCreators/web-stories-wp/pull/10511#discussion_r801352022

Basically, this package does not yet provide a list of constants, but most likely will in a future version

Reply:
That comment thread was hidden so didnt see this. My comment about adding a todo / comment is still valid.

Reply:
I'll add a comment ... there's an issue filed here -- it's currently an empty file.

https://github.com/sniccowp/php-scoper-wordpress-excludes/issues/2